src/util.ts

```diff
@@ -32,3 +38,47 @@ export const randomString = (length = 6, adjust = 0): string => {
 
 export const sleep = (ms: number): Promise<void> =>
   new Promise((resolve) => setTimeout(resolve, ms));
+
+/**
+ * Waits for the middleware to sync up to the block containing the given transaction
+ *
+ * @param tx - The transaction to wait for
+ * @param sdk - The Polymesh SDK instance
+ * @param retries - The number of times to retry checking the middleware
+ * @param delay - The delay in milliseconds between retries
+ * @throws {Error} If the transaction status is not Succeeded or Failed
+ * @throws {Error} If the middleware has not synced after the specified number of retries
+ * @returns {Promise<void>} Resolves when the middleware has synced up to the transaction's block
+ */
+export const awaitMiddlewareSynced = async (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  tx: GenericPolymeshTransaction<any, any>,
+  sdk: Polymesh,
+  retries: number,
+  delay: number
```

Comment:
What about default values so we don't have to repeat them?

```suggestion
  retries = 15,
  delay = 2000
```